python/tk_framework_desktopserver/server_protocol.py

```diff
@@ -89,15 +89,20 @@ def connectionLost(self, reason):
             # Known certificate error. These work for firefox and safari, but chrome rejected certificate
             # are currently indistinguishable from lost connection. This is true as of July 21st, 2015.
             certificate_error = False
-            certificate_error |= (
-                reason.type is OpenSSL.SSL.Error
-                and reason.value.message[0][2] == "ssl handshake failure"
-            )
-            certificate_error |= (
-                reason.type is OpenSSL.SSL.Error
-                and reason.value.message[0][2] == "tlsv1 alert unknown ca"
-            )
+            try:
+                certificate_error |= (
```

Comment:
Is it possible to use only one line to get message? :
`message = reason.value.message[0][2]`
and then use "message" in comparisons?

Reply:
I agree, could you extract the message in a small try/except block and then use it:

```python
try:
  message = reason.value.message[0][2]
except AttributeError, IndexError:
  logger.warning("Unexpected error message in object: {}".format(reason))
else:
  certificate_error |= (
    reason.type is OpenSSL.SSL.Error
    and message == "ssl handshake failure"
  )
  certificate_error |= (
    reason.type is OpenSSL.SSL.Error
    and message == "tlsv1 alert unknown ca"
  )
```

Reply:
Sounds good. My intention wasn't to do a full refactor but I can take the chance.